pyccel/codegen/printing/ccode.py

```diff
@@ -1760,7 +1760,9 @@ def _print_PyccelArrayShapeElement(self, expr):
     def _print_Allocate(self, expr):
         free_code = ''
         variable = expr.variable
-        if isinstance(variable.class_type, (HomogeneousListType, HomogeneousSetType, DictType)):
+        if isinstance(variable.class_type, (HomogeneousListType, HomogeneousSetType, DictType, StringType)):
+            if expr.status in ('allocated', 'unknown'):
+                free_code = f'{self._print(Deallocate(variable))}'
             if expr.shape[0] is None:
                 return free_code
             if expr.status in ('allocated', 'unknown'):
```

Comment:
Careful it looks like you have duplicated `if expr.status in ('allocated', 'unknown'):` in this merge